main/autogen/autogen.cc

```diff
@@ -711,9 +709,22 @@ void ParsedFile::subclasses(core::Context ctx, map<string, set<string>> &out) {
         string parentName = fmt::format("{}", fmt::map_join(ref.resolved, "::", nameToString));
 
         // Add child class to the set identified by its parent
-        string childName = fmt::format("{},{}", fmt::map_join(showFullName(ctx, defn), "::", nameToString),
-                                       defn.data(*this).typeAsStringView());
-        out[parentName].insert(childName);
+        string childName = fmt::format("{}", fmt::map_join(showFullName(ctx, defn), "::", nameToString));
+
+        out[parentName].insert(make_pair(childName, defn.data(*this).type));
+    }
+}
+
+void descendantsOf(map<string, set<pair<string, Definition::Type>>> &childMap, string &parentName,
+                   set<pair<string, Definition::Type>> &out) {
```

Comment:
Considering making a type alias for <pair<string, Definition::Type>> 'cuz I use it everywhere

Reply:
[added](https://github.com/sorbet/sorbet/pull/1040/files#diff-670beace6f2de3dc49c6286fb9310f27R74)